dstore.go

```diff
@@ -75,7 +73,6 @@ func (b *batch) Put(key ds.Key, val []byte) error {
 func (b *batch) Delete(key ds.Key) error {
 	txn, err := b.GetTransaction()
 	if err != nil {
-		_ = b.txn.Rollback()
```

Comment:
ℹ️ It looks like this changes semantics from "on conflict rollback" to "on conflict abort" https://sqlite.org/lang_conflict.html

That looks right - at least the badger implementation uses "on conflict abort" semantics - but we should probably document that more clearly in go-datastore

Reply:
`GetTransaction()` rolls back the transaction if it can. I removed this line because of this, and the fact that `b.txn` might be `nil` at this point if an error occurred.